common/containers/Tabs/Contracts/components/Deploy.tsx

```diff
@@ -46,27 +45,12 @@ class DeployClass extends Component<DispatchProps> {
           />
         </div>
 
-        <label className="Deploy-field form-group">
-          <h4 className="Deploy-field-label">Gas Limit</h4>
-          <GasLimitFieldFactory
-            withProps={({ gasLimit: { raw, value }, onChange, readOnly }) => (
-              <input
-                name="gasLimit"
-                value={raw}
-                disabled={readOnly}
-                onChange={onChange}
-                className={classnames('Deploy-field-input', 'form-control', {
-                  'is-invalid': !value
-                })}
-              />
-            )}
-          />
-        </label>
         <div className="row form-group">
-          <div className="col-xs-11">
-            <NonceField alwaysDisplay={false} />
+          <div className="col-xs-12 clearfix">
+            <GasSlider />
```

Comment:
Same as interact, should have a `hideData={true}` prop since the textarea will be the data.